samples/csharp_dotnetcore/80.skills-simple-bot-to-bot/SimpleRootBot/AdapterWithErrorHandler.cs

```diff
@@ -13,42 +18,96 @@ namespace Microsoft.BotBuilderSamples.SimpleRootBot
 {
     public class AdapterWithErrorHandler : BotFrameworkHttpAdapter
     {
-        public AdapterWithErrorHandler(IConfiguration configuration, ILogger<BotFrameworkHttpAdapter> logger, ConversationState conversationState = null)
+        private ConversationState _conversationState;
+        private IConfiguration _configuration;
+        private ILogger _logger;
+        private SkillHttpClient _skillClient;
+        private SkillsConfiguration _skillsConfig;
+
+        public AdapterWithErrorHandler(IConfiguration configuration, ILogger<BotFrameworkHttpAdapter> logger, ConversationState conversationState = null, SkillHttpClient skillClient = null, SkillsConfiguration skillsConfig = null)
             : base(configuration, logger)
         {
-            OnTurnError = async (turnContext, exception) =>
+            _configuration = configuration;
+            _conversationState = conversationState;
+            _logger = logger;
+            _skillClient = skillClient;
+            _skillsConfig = skillsConfig;
+
+            OnTurnError = OnBotError;
+        }
+
+        private async Task OnBotError(ITurnContext turnContext, Exception exception)
+        {
+            // Log any leaked exception from the application.
+            _logger.LogError(exception, $"[OnTurnError] unhandled error : {exception.Message}");
```

Comment:
Replace by the following to ensure logger is not null:

``
_logger = logger ?? throw new ArgumentNullException(nameof(logger));;
``